### ISSUE TITLE
Do not arbitrarily vary parameter names in overloads

### ISSUE THREAD

**Issue by RalfKoban** [RalfKoban/MiKo-Analyzers#669]
We should report a warning if the parameter names in the overloads do not match.

The reason is that most likely these parameters represent the same inputs across the different overloads, so they should have the same name.

Example:
```c#
public void DoSomething(int i);
public void DoSomething(int j, int k);
public void DoSomething(int m, int n, int o);
```

It seems like the parameters are all different ones, but actually they are the same input. So they should be named the same:
```c#
public void DoSomething(int i);
public void DoSomething(int i, int j);
public void DoSomething(int i, int j, int k);
```

This is related to #759
